src/System.Private.CoreLib/src/System/Threading/WaitHandle.cs

```diff
@@ -26,13 +26,10 @@ public abstract class WaitHandle : MarshalByRefObject, IDisposable
 
         private const int MAX_WAITHANDLES = 64;
 
-#pragma warning disable 414  // Field is not used from managed.
-        private IntPtr waitHandle;  // !!! DO NOT MOVE THIS FIELD. (See defn of WAITHANDLEREF in object.h - has hard-coded access to this field.)
-#pragma warning restore 414
+        // IMPORTANT:
+        // - Do not add or rearrange fields as the EE depends on this layout.
 
-        internal volatile SafeWaitHandle _waitHandle;
-
-        internal bool hasThreadAffinity;
+        internal SafeWaitHandle _waitHandle;
```

Comment:
nit: Would be nice to remove OpenExistingResult

Reply:
Will do that in a follow up PR (replacing the code with CoreRT-based version and moving to shared). I wanted to tackle the changes to unmanaged code separately to make it easier to review.